client/extensions/woocommerce/state/sites/orders/activity-log/selectors.js

```diff
@@ -0,0 +1,143 @@
+		( ! config.isEnabled( 'woocommerce/extension-wcservices' ) || areShippingLabelsLoaded( state, orderId, siteId ) );
+};
+
+/**
+ * @param {Object} state Whole Redux state tree
+ * @param {Number} orderId Order ID to check.
+ * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
+ * @return {boolean} Whether the activity log for a given order is currently being retrieved from the server.
+ */
+export const isActivityLogLoading = ( state, orderId, siteId = getSelectedSiteId( state ) ) => {
+	if ( config.isEnabled( 'woocommerce/extension-wcservices' ) ) {
+		return areOrderNotesLoading( state, orderId, siteId ) || areShippingLabelsLoading( state, orderId, siteId );
+	}
+	return areOrderNotesLoading( state, orderId, siteId );
+};
+
+/**
+ * @param {Object} state Whole Redux state tree
+ * @param {Number} orderId Order ID to check.
+ * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
+ * @return {Object[]} List of events to display. Each event will have at least these properties:
+ * - {String} type The type of the event. See the EVENT_TYPES enum.
+ * - {Number} key A unique ID for the event. The combination of "type + key" must be unique in the whole list.
+ * - {Number} timestamp The time of the event.
+ */
+export const getActivityLogEvents = ( state, orderId, siteId = getSelectedSiteId( state ) ) => {
+	const events = getOrderNotes( state, orderId, siteId ).map( note => ( {
+		key: note.id,
+		type: note.customer_note ? EVENT_TYPES.CUSTOMER_NOTE : EVENT_TYPES.INTERNAL_NOTE,
+		timestamp: new Date( note.date_created_gmt + 'Z' ).getTime(),
+		content: note.note,
+	} ) );
+
+	if ( config.isEnabled( 'woocommerce/extension-wcservices' ) ) {
+		getLabels( state, orderId, siteId ).forEach( ( label, index ) => {
+			if ( label.refund ) {
+				switch ( label.refund.status ) {
+					case 'complete':
+						events.push( {
+							key: label.label_id,
+							type: EVENT_TYPES.LABEL_REFUND_COMPLETED,
+							timestamp: label.refund.refund_date,
+							labelIndex: index,
+							amount: parseFloat( label.refund.amount ),
+							currency: label.currency,
+						} );
+						break;
+					case 'rejected':
+						events.push( {
+							key: label.label_id,
+							type: EVENT_TYPES.LABEL_REFUND_REJECTED,
+							timestamp: label.refund.refund_date,
+							labelIndex: index,
+						} );
+						break;
+					default:
+						// Only render the "refund requested" event if the refund hasn't yet completed/rejected
+						events.push( {
+							key: label.label_id,
+							type: EVENT_TYPES.LABEL_REFUND_REQUESTED,
+							timestamp: label.refund.request_date,
+							labelIndex: index,
+							amount: parseFloat( label.refund.amount ) || label.refundable_amount,
+							currency: label.currency,
+						} );
+				}
+			}
+			events.push( {
+				key: label.label_id,
+				type: EVENT_TYPES.LABEL_PURCHASED,
+				timestamp: label.created_date,
+				labelIndex: index,
```

Comment:
The labels have been numbered as `labels.length - index` to this point https://github.com/Automattic/woocommerce-services/blob/a8974547f3867ebcad0015b865bb76e558266538/client/apps/shipping-label/view.js#L98
Do we want to stick with that?

Reply:
Yes we do, great catch! The server replies with the last label first, so we must flip the index here.
Done in https://github.com/Automattic/wp-calypso/pull/17807/commits/1cce3554757dafd4f81168d35b795f456149c162